### PR TITLE
Feature/update csv format

### PR DESCRIPTION
**Pull request by redhug1**
### What

Move and rename count column.
Fix many unhandled errors and othe code issues.

### How to review

Visual inspection.
Run tests.

### Who can review

Anyone

### CODE REVIEW

File: filter/filter.go [hunk 1]
```diff
@@ -165,9 +164,18 @@ func (c *Client) UpdateFilterOutputBytes(ctx context.Context, userAuthToken, ser
...
 	}
 
-	headers.SetAuthToken(req, userAuthToken)
-	headers.SetServiceAuthToken(req, serviceAuthToken)
-	headers.SetDownloadServiceToken(req, downloadServiceToken)
+	err = headers.SetAuthToken(req, userAuthToken)
+	if err != nil {
+		return err
+	}
+	err = headers.SetServiceAuthToken(req, serviceAuthToken)
```
Comment (ChrysmOre) [inline filter/filter.go:171]: Can use `if err = .. ` shorthand here on these lines and below

Comment (ChrysmOre) [reply filter/filter.go:171]: Also could you please wrap each of these errors (and basically all errors you return ever!) with fmt.Errorf and brief descriptive context? i.e.:
`return fmt.Errorf("failed to set download service token header: %w", err)`

**Review by ChrysmOre — CHANGES_REQUESTED** [review]
Couple of questions and error wrapping

File: cantabular/client.go [hunk 2]
```diff
@@ -84,28 +84,26 @@ func (c *Client) Checker(ctx context.Context, state *healthcheck.CheckState) err
 		"service": Service,
 	}
 	code := 0
-	url := fmt.Sprintf("%s/v9/datasets", c.host)
+	urlHost := fmt.Sprintf("%s/v9/datasets", c.host)
 
-	res, err := c.httpGet(ctx, url)
+	res, err := c.httpGet(ctx, urlHost)
```
[the remark this answers is not in the document]
Comment (redhug1) [reply cantabular/client.go:89]: The reason for changing it is:
Variable 'url' collides with imported package name
have changed it to: 'hostPath'

File: cantabular/table.go [hunk 3]
```diff
@@ -17,24 +17,24 @@ type Table struct {
...
 	b := new(bytes.Buffer)
 	w := csv.NewWriter(b)
 
-	// Create and write header seperately
+	// Create and write header separately
 	header := c.createCSVHeader(table.Dimensions)
 
-	w.Write(header)
-	if err := w.Error(); err != nil {
+	err := w.Write(header)
```
[the remark this answers is not in the document]
Comment (redhug1) [reply cantabular/table.go:28]: fixed.

File: filter/filter.go (hunk 1 above)
Comment (redhug1) [reply filter/filter.go:171]: FYI i was just fixing the unhandled errors as a first pass to just make the code more robust ...

File: cantabular/client.go [hunk 4]
```diff
@@ -84,9 +84,9 @@ func (c *Client) Checker(ctx context.Context, state *healthcheck.CheckState) err
 		"service": Service,
 	}
 	code := 0
-	urlHost := fmt.Sprintf("%s/v9/datasets", c.host)
+	hostPath := fmt.Sprintf("%s/v9/datasets", c.host)
```
Comment (ChrysmOre) [inline cantabular/client.go:87]: Sorry I think hostPath is still a bit ambiguous.. Really `url` is fine, it's a really common naming convention for this situation

File: cantabular/codebook.go [hunk 5]
```diff
@@ -2,38 +2,38 @@ package cantabular
...
 type Codebook []Variable
 
 // GetCodebook gets a Codebook from cantabular.
-func (c *Client) GetCodebook(ctx context.Context, req GetCodebookRequest) (*GetCodebookResponse, error){
-	if len(c.host) == 0{
+func (c *Client) GetCodebook(ctx context.Context, req GetCodebookRequest) (*GetCodebookResponse, error) {
+	if len(c.host) == 0 {
 		return nil, dperrors.New(
-			errors.New("Cantabular Server host not configured"),
+			errors.New("cantabular Server host not configured"),
```
[the remark this answers is not in the document]
Comment (redhug1) [reply cantabular/codebook.go:22]: Ta. fixed.

File: dataset/dataset.go [hunk 6]
```diff
@@ -713,7 +713,7 @@ func (c *Client) GetInstances(ctx context.Context, userAuthToken, serviceAuthTok
 		return
 	}
 
-	json.Unmarshal(b, &m)
+	err = json.Unmarshal(b, &m)
```
[the remark this answers is not in the document]
Comment (redhug1) [reply dataset/dataset.go:716]: To capture and process the unhandled error in the original code.

File: codelist/codelist.go [hunk 7]
```diff
@@ -310,12 +310,11 @@ func setAuthenticationHeaders(req *http.Request, userAuthToken, serviceAuthToken
...
 
+// closeResponseBody closes the response body and logs an error if unsuccessful
 func closeResponseBody(ctx context.Context, resp *http.Response) {
-	if resp.Body == nil {
-		return
-	}
-
-	if err := resp.Body.Close(); err != nil {
-		log.Error(ctx, "error closing http response body", err)
+	if resp.Body != nil {
```
[the remark this answers is not in the document]
Comment (redhug1) [reply codelist/codelist.go:315]: Thank you

[the remark this answers is not in the document]
Comment (redhug1) [reply codelist/codelist.go:315]: Thank you

File: cantabular/client.go (hunk 4 above)
Comment (redhug1) [reply cantabular/client.go]: Adjusted to `reqURL` as per discussion elsewhere.

**Review by ChrysmOre — APPROVED** [review]

**Pull request merged**